promql/engine.go

```diff
@@ -1199,13 +1203,8 @@ func (ev *evaluator) rangeEval(prepSeries func(labels.Labels, *EvalSeriesHelper)
 				if prepSeries != nil {
 					bufHelpers[i] = append(bufHelpers[i], seriesHelpers[i][si])
 				}
-				ev.currentSamples++
```

Comment:
Or maybe not, because we don't `Copy` the histogram, just the pointer to it. In that case, we need a comment here. Will think more about it.

Reply:
Yeah, pretty sure the current state is correct because we only copy the pointer. I'll add a code comment here in a PR that I'm already preparing for related issues.

Reply:
```diff
- ev.currentSamples++
```
this line is deleted because we have already accounted all the samples loaded in line [1126](https://github.com/prometheus/prometheus/pull/9626/files#diff-ad2761bb162a2f12326b18f2714ef54f4cd476cfe3388f77b2b20c5ff82e4faaR1126)